lib/print.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * print.c - common print support functions for lsof
+ */
+
+/*
+ * Copyright 1994 Purdue Research Foundation, West Lafayette, Indiana
+ * 47907.  All rights reserved.
+ *
+ * Written by Victor A. Abell
+ *
+ * This software is not subject to any license of the American Telephone
+ * and Telegraph Company or the Regents of the University of California.
+ *
+ * Permission is granted to anyone to use this software for any purpose on
+ * any computer system, and to alter it and redistribute it freely, subject
+ * to the following restrictions:
+ *
+ * 1. Neither the authors nor Purdue University are responsible for any
+ *    consequences of the use of this software.
+ *
+ * 2. The origin of this software must not be misrepresented, either by
+ *    explicit claim or by omission.  Credit to the authors and Purdue
+ *    University must appear in documentation and sources.
+ *
+ * 3. Altered versions must be plainly marked as such, and must not be
+ *    misrepresented as being the original software.
+ *
+ * 4. This notice may not be removed or altered.
+ */
+
+#include "common.h"
+#include "lsof.h"
+
+/*
+ * Local definitions, structures and function prototypes
+ */
+
+/*
+ * print_access() - print enum lsof_file_access_mode
+ */
+char print_access(enum lsof_file_access_mode access) {
```

Comment:
The naming for `print_access` is a bit misleading as this function determines the character used to indicate the kind of access on the file; it doesn't actually print anything.

Also it might be worth distinguishing the `LSOF_FILE_ACCESS_NONE` case and undefined by returning `?` for the default case. This should not happen, but better provide this distinction then silently drop it …

Reply:
The space is expected to fill one character's width in the tabular output.